css-shared-element-transitions/index.bs

```diff
@@ -298,241 +299,302 @@ Single-page API:
-        [=abandon the page transition=] managed by [=this=]
-        with an {{AbortError}}.
+    1. If [=this's=] [=SameDocumentTransition/phase=] is not "`finished`",
+        then [=abandon the page transition=] [=this=]
+        with an "{{AbortError}}" {{DOMException}}.
+</div>
+
+<div algorithm="monkey patch rendering">
+    Run the following steps before [marking paint timing](https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-processing-model:mark-paint-timing) in the [=update the rendering=] steps:
+
+    1. For each [=fully active=] {{Document}} in <var ignore>docs</var>,
+        [=perform pending transition operations=] for that {{Document}}.
+
+    Note: These steps will be added to the [=update the rendering=] in the HTML spec.
+        As such, the prose style is written to match other steps in that algorithm.
+</div>
+
+<div algorithm="perform pending transition operations">
+    To <dfn>perform pending transition operations</dfn> given a {{Document}} |document|,
+        perform the following steps:
+
+    1. If |document|'s [=document/pending same document transition outgoing capture=] is not null, then:
+
+        1. [=Perform an outgoing capture=] with |document|'s [=document/pending same document transition outgoing capture=].
+
+        1. Set |document|'s [=document/pending same document transition outgoing capture=] to null.
+
+    1. If |document|'s [=document/running same document transition=] is not null,
+        then [=update transition DOM=] for |document|'s [=document/running same document transition=].
+</div>
+
+<div algorithm="perform outgoing capture">
+    To <dfn>perform an outgoing capture</dfn> given a {{SameDocumentTransition}} |transition|,
+        perform the following steps:
+
+    1. Let |taggedElements| be |transition|'s [=SameDocumentTransition/tagged elements=].
+
+    1. Let |usedTransitionTags| be a new [=/set=] of strings.
+
+    1. Let |preparationFailed| be false.
+
+    1. Let |document| be |transition|'s [=relevant global object's=] [=associated document=].
+
+    1. [=list/For each=] |element| of every {{Element}} and [=pseudo-element=] connected to |document|,
+        in [paint order](https://drafts.csswg.org/css2/#painting-order):
+
+        Issue: The link for "paint order" doesn't seem right.
+            Is there a more canonical definition?
+
+        1. Let |transitionTag| be the [=computed value=] of 'page-transition-tag' for |element|.
+
+        1. If |transitionTag| is ''page-transition-tag/none'',
+            or |element| is [=element-not-rendered|not rendered=],
+            then [=continue=].
+
+        1. If any of the following is true:
+
+            * |usedTransitionTags| [=list/contains=] |transitionTag|.
+
+            * |element| is not |element|'s [=tree/root=] and |element| does not have [=layout containment=].
+
+            * |element| is not |element|'s [=tree/root=] and |element| does prevent [=fragmentation=].
```

Comment:
```suggestion
            * |element| is not |element|'s [=tree/root=] and |element| allows [=fragmentation=].
```

We need to ensure the content is not fragmented to make it easier to snapshot. We need an issue to clearly define what the developer needs to do in this case. Like adding "break-inside:avoid" to the element's style.

Reply:
Good catch. I missed a "not" there, but "allows" is better.